webob/exc.py

```diff
@@ -308,6 +322,9 @@ def generate_response(self, environ, start_response):
         if accept and 'html' in accept or '*/*' in accept:
             content_type = 'text/html'
             body = self.html_body(environ)
+        elif accept and 'json' in accept:
```

Comment:
If you have an XHR request that also still has `*/*` in it, this case will not get hit. Is this acceptable? Or do we want to see if JSON is listed first before checking for `HTML`/`*/*`?